Wikipedia/Code/TalkPageFetcher.swift

```diff
@@ -115,19 +104,28 @@ class TalkPageFetcher: Fetcher {
         
         performTokenizedMediaWikiAPIPOST(to: siteURL, with: params, reattemptLoginOn401Response: false) { result, httpResponse, error in
             
-            if let error = error {
-                completion(.failure(error))
-            }
-            
-            if let resultError = result?["error"] as? [String: Any],
-               let info = resultError["info"] as? String {
-                completion(.failure(RequestError.api(info)))
-                return
-            }
-            
-            if let resultSuccess = result?["success"] as? [String: Any] {
-                completion(.success(resultSuccess))
-            }
+            self.performTalkPagePost(error, result, completion: completion)
         }
     }
+    
+    fileprivate func performTalkPagePost(_ error: Error?, _ result: [String : Any]?, completion: @escaping(Result<[AnyHashable: Any], Error>) -> Void) {
+        if let error = error {
+            completion(.failure(error))
+            return
+        }
+        
+        if let resultError = result?["error"] as? [String: Any],
+           let info = resultError["info"] as? String {
+            completion(.failure(RequestError.api(info)))
+            return
+        }
+        
+        if let resultSuccess = result?["discussiontoolsedit"] as? [String: Any] {
```

Comment:
I think it'd still be good to check for the "success" value here, like you were before. Also I don't think we'll need to pass back / access the dictionary afterwards, just a basic success flag should be enough. You can use `Void` in Swift's `Result` success value to avoid needing to pass something back:

```
fileprivate func performTalkPagePost(_ error: Error?, _ result: [String : Any]?, completion: @escaping(Result<Void, Error>) -> Void) {
```

Then here it could be something like (using `guard` because I think that's a bit more clear):

```
        guard let discussionToolsEdit = result?["discussiontoolsedit"] as? [String: Any],
              let discussionToolsEditResult = discussionToolsEdit["result"] as? String,
              discussionToolsEditResult == "success" else {
                completion(.failure(RequestError.unexpectedResponse))
            return
        }
        
        completion(.success(()))
```